arches/app/etl_modules/branch_excel_importer.py

```diff
@@ -203,8 +203,11 @@ def download(self, request):
         format = request.POST.get("format")
         if format == "xls":
             wb = create_workbook(request.POST.get("id"))
-            response = HttpResponse(save_virtual_workbook(wb), content_type="application/vnd.ms-excel")
-            response["Content-Disposition"] = "attachment"
-            return {"success": True, "raw": response}
+            with NamedTemporaryFile(suffix='.xlsx', delete=False) as tmp:
```

Comment:
Thanks for the `delete=False` update. I think to tidy up afterward you'll also want an `os.unlink(tmp)` after the with block (and before returning), as otherwise the file will remain.

(The intent of `delete=False` is not to leave the files behind afterward, but just to ensure `wb.save()` doesn't fail on Windows as described [here](https://github.com/archesproject/arches/pull/10322#issue-2021167595).)

Reply:
Do you think it's necessary to add try/except in the event of an error to be seen in the logs?

Reply:
I'm not sure if it's necessary to add the try block, but I don't see any harm in it. In any case, it would still be important to add an `os.unlink(tmp)`. Sorry I didn't catch that detail earlier.

Reply:
I'll leave it as is in that case. It can be modified if necessary. I'm the one who didn't read the issue carefully enough.